### ISSUE TITLE
Old flutter gallery text contrast tests fail when images actually load

### ISSUE THREAD

**Issue by jonahwilliams** [flutter/flutter#103669]
```
00:27 +85: /opt/s/w/ir/x/w/flutter/dev/integration_tests/flutter_gallery/test/accessibility_test.dart: Gallery home page meets text contrast guidelines options menu - dark theme                      
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following TestFailure was thrown running a test:
Expected: Text contrast should follow WCAG guidelines
  Actual: <Instance of 'WidgetTester'>
   Which: SemanticsNode#35(merge boundary ⛔️, Rect.fromLTRB(0.0, 0.0, 800.0, 48.0), flags:
[isHeader], label: "Diagnostics", textDirection: ltr):
          Expected contrast ratio of at least 3.0 but found 1.92 for a font size of 20.0.
          The computed colors was:
          light - Color(0xff003d75), dark - Color(0xff000000)
          See also: https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-contrast.html

When the exception was thrown, this was the stack:
#0      fail (package:test_api/src/expect/expect.dart:137:31)
#1      _expect.<anonymous closure> (package:test_api/src/expect/expect.dart:112:9)
<asynchronous suspension>
<asynchronous suspension>
(elided 6 frames from dart:async and package:stack_trace)

The test description was:
  options menu - dark theme
════════════════════════════════════════════════════════════════════════════════════════════════════

00:27 +85 -1: /opt/s/w/ir/x/w/flutter/dev/integration_tests/flutter_gallery/test/accessibility_test.dart: Gallery home page meets text contrast guidelines options menu - dark theme [E]               
  Test failed. See exception logs above.
  The test description was: options menu - dark theme
  
  ```

**Comment by jonahwilliams** [comment]
Due to a misunderstanding of https://github.com/flutter/flutter/issues/103724 this issue was raised when we tested with real icons instead of ahem
